pkg/networkserver/grpc_mac_settings_profile.go

```diff
@@ -161,13 +165,17 @@ func (m *NsMACSettingsProfileRegistry) List(ctx context.Context, req *ttnpb.List
 	if req.FieldMask != nil {
 		paths = req.FieldMask.GetPaths()
 	}
+
+	var total int64
+	ctx = m.registry.WithPagination(ctx, req.Limit, req.Page, &total)
```

Comment:
Question: if `req.Limit` is 0, what happens? On first glance it looks like no limit is enforced, while we always should enforce a limit.

Reply:
If `req.Limit` is 0 then we overwriting with the default limit, so we always enforcing the default limit.

```
func NewContextWithPagination(ctx context.Context, limit, page int64, total *int64) context.Context {
	if page == 0 {
		page = 1
	}
	if limit == 0 {
		limit = paginationDefaults.DefaultLimit
	}
	return context.WithValue(ctx, paginationOptionsKey, paginationOptions{
		limit:  limit,
		offset: (page - 1) * limit,
		total:  total,
	})
}
```